src/Test.Utilities/CSharpCodeFixVerifier`2.cs

```diff
@@ -23,14 +24,29 @@ public static DiagnosticResult Diagnostic(string diagnosticId)
         public static DiagnosticResult Diagnostic(DiagnosticDescriptor descriptor)
             => CSharpCodeFixVerifier<TAnalyzer, TCodeFix, XUnitVerifier>.Diagnostic(descriptor);
 
-        public static async Task VerifyAnalyzerAsync(string source, params DiagnosticResult[] expected)
+        public static Task VerifyAnalyzerAsync(string source, params DiagnosticResult[] expected)
+            => VerifyAnalyzerAsync(source, expectedDiagnostics: expected);
+
+        public static Task VerifyAnalyzerAsync(string source, CompilerDiagnostics compilerDiagnostics, params DiagnosticResult[] expected)
+            => VerifyAnalyzerAsync(source, expected, compilerDiagnostics);
+
+        public static Task VerifyAnalyzerWithEditorConfigAsync(string source, string editorConfigSource, params DiagnosticResult[] expected)
+            => VerifyAnalyzerAsync(source, expected, additionalFiles: new SourceFileCollection { (".editorconfig", SourceText.From(editorConfigSource)) });
+
+        public static async Task VerifyAnalyzerAsync(string source, DiagnosticResult[] expectedDiagnostics,
+            CompilerDiagnostics compilerDiagnostics = CompilerDiagnostics.Errors, SourceFileCollection additionalFiles = null)
```

Comment:
Typically, it is preferred to write the test using the "long form" rather than add overloads to this class. For example, a test using both `compilerDiagnostics` and **.editorconfig** could be written like this:

```csharp
await new VerifyCS.Test
{
  TestState =
  {
    Sources = { source },
    AdditionalFiles = { (".editorconfig", editorConfig) },
  },
  CompilerDiagnostics = CompilerDiagnostics.Errors,
  ExpectedDiagnostics = { diagnostic },
}.RunAsync();
```

Reply:
Shall I apply the same for the overload with the `CompilerDiagnostics` enum parameter?

Reply:
➡️ Will be addressed as part of #2995